lib/bots/versionbot.ts

```diff
@@ -764,14 +769,32 @@ export class VersionBot extends ProcBot {
         //     `package.json` file. This means components that don't have one need a custom
         //     `versionist.conf.js` in their root dir. And we need to test to run against it.
         //     It's possible to get round this using a custom `versionist.conf.js`, which we now support.
-        const cliCommand = (command: string) => {
-            return exec(command, { cwd: versionData.fullPath });
+        const cliCommand = (params: ExternalCommand) => {
+            return new Promise((resolve, reject) => {
+                const child = spawn(params.command, params.args,
+                    (params.workingDir) ? { cwd: params.workingDir } : undefined);
+                let stdout = '';
+
+                child.stdout.on('data', (data) => {
+                    stdout += data.toString();
+                });
+                child.addListener('close', () => resolve(stdout));
+                child.addListener('error', reject);
+            });
         };
 
         return Promise.mapSeries([
-            `git clone https://${versionData.authToken}:${versionData.authToken}@github.com/` +
-                `${versionData.repoFullName} ${versionData.fullPath}`,
-            `git checkout ${versionData.branchName}`
+            {
+                command: 'git',
+                args: ['clone', `https://${versionData.authToken}:${versionData.authToken}@github.com/` +
+                `${versionData.repoFullName}`, `${versionData.fullPath}`],
+                workingDir: `${versionData.fullPath}`
+            },
+            {
+                command: 'git',
+                args: ['checkout', `${versionData.branchName}`],
+                workingDir: `${versionData.fullPath}`
+            }
```

Comment:
Could make this a little more readable with a command that builds these objects:

```typescript
buildCommand('git', ['clone', 'https://...'])
```